### ISSUE TITLE
Edge Functions break after upgrading to canary.12 with TS2345 [ERROR]

### ISSUE THREAD

**Issue by stevenmilstein** [11ty/eleventy#2487]
**Describe the bug**
Apologies! I don't know how to reproduce this error and my repository is private. What I do know is that after upgrading from canary.11 to canary.12 and running `netlify dev` locally, I receive the following in the console after a non-Edge page reloads:
```
error: TS2345 [ERROR]: Argument of type '"CAA"' is not assignable to parameter of type 'RecordType'.
        this.#query(name, "CAA").then(({ ret }) => {
                          ~~~~~
    at https://deno.land/std@0.144.0/node/internal_binding/cares_wrap.ts:266:27

TS2694 [ERROR]: Namespace 'Deno' has no exported member 'CAARecord'.
          (ret as Deno.CAARecord[]).forEach(({ critical, tag, value }) =>
                       ~~~~~~~~~
    at https://deno.land/std@0.144.0/node/internal_binding/cares_wrap.ts:267:24

TS2322 [ERROR]: Type '"CAA"' is not assignable to type 'RecordType'.
              type: "CAA",
              ~~~~
    at https://deno.land/std@0.144.0/node/internal_binding/cares_wrap.ts:269:15

    The expected type comes from property 'type' which is declared here on type '{ [key: string]: unknown; type: RecordType; }'
          const records: { type: Deno.RecordType; [key: string]: unknown }[] = [];
                           ~~~~
        at https://deno.land/std@0.144.0/node/internal_binding/cares_wrap.ts:255:24

TS2345 [ERROR]: Argument of type '"NAPTR"' is not assignable to parameter of type 'RecordType'.
        this.#query(name, "NAPTR").then(({ ret }) => {
                          ~~~~~~~
    at https://deno.land/std@0.144.0/node/internal_binding/cares_wrap.ts:289:27

TS2694 [ERROR]: Namespace 'Deno' has no exported member 'NAPTRRecord'.
          (ret as Deno.NAPTRRecord[]).forEach(
                       ~~~~~~~~~~~
    at https://deno.land/std@0.144.0/node/internal_binding/cares_wrap.ts:290:24

TS2322 [ERROR]: Type '"NAPTR"' is not assignable to type 'RecordType'.
                type: "NAPTR",
                ~~~~
    at https://deno.land/std@0.144.0/node/internal_binding/cares_wrap.ts:293:17

    The expected type comes from property 'type' which is declared here on type '{ [key: string]: unknown; type: RecordType; }'
          const records: { type: Deno.RecordType; [key: string]: unknown }[] = [];
                           ~~~~
        at https://deno.land/std@0.144.0/node/internal_binding/cares_wrap.ts:255:24

TS2345 [ERROR]: Argument of type '"NS"' is not assignable to parameter of type 'RecordType'.
        this.#query(name, "NS").then(({ ret }) => {
                          ~~~~
    at https://deno.land/std@0.144.0/node/internal_binding/cares_wrap.ts:303:27

TS2322 [ERROR]: Type '"NS"' is not assignable to type 'RecordType'.
            records.push({ type: "NS", value: fqdnToHostname(record) })
                           ~~~~
    at https://deno.land/std@0.144.0/node/internal_binding/cares_wrap.ts:305:28

    The expected type comes from property 'type' which is declared here on type '{ [key: string]: unknown; type: RecordType; }'
          const records: { type: Deno.RecordType; [key: string]: unknown }[] = [];
                           ~~~~
        at https://deno.land/std@0.144.0/node/internal_binding/cares_wrap.ts:255:24

TS2345 [ERROR]: Argument of type '"SOA"' is not assignable to parameter of type 'RecordType'.
        this.#query(name, "SOA").then(({ ret }) => {
                          ~~~~~
    at https://deno.land/std@0.144.0/node/internal_binding/cares_wrap.ts:313:27

TS2694 [ERROR]: Namespace 'Deno' has no exported member 'SOARecord'.
          (ret as Deno.SOARecord[]).forEach(
                       ~~~~~~~~~
    at https://deno.land/std@0.144.0/node/internal_binding/cares_wrap.ts:314:24

TS2322 [ERROR]: Type '"SOA"' is not assignable to type 'RecordType'.
                type: "SOA",
                ~~~~
    at https://deno.land/std@0.144.0/node/internal_binding/cares_wrap.ts:317:17

    The expected type comes from property 'type' which is declared here on type '{ [key: string]: unknown; type: RecordType; }'
          const records: { type: Deno.RecordType; [key: string]: unknown }[] = [];
                           ~~~~
        at https://deno.land/std@0.144.0/node/internal_binding/cares_wrap.ts:255:24

TS2345 [ERROR]: Argument of type '"CAA"' is not assignable to parameter of type 'RecordType'.
    this.#query(name, "CAA").then(({ code, ret }) => {
                      ~~~~~
    at https://deno.land/std@0.144.0/node/internal_binding/cares_wrap.ts:374:23

TS2694 [ERROR]: Namespace 'Deno' has no exported member 'CAARecord'.
      const records = (ret as Deno.CAARecord[]).map(
                                   ~~~~~~~~~
    at https://deno.land/std@0.144.0/node/internal_binding/cares_wrap.ts:375:36

TS2345 [ERROR]: Argument of type '"NAPTR"' is not assignable to parameter of type 'RecordType'.
    this.#query(name, "NAPTR").then(({ code, ret }) => {
                      ~~~~~~~
    at https://deno.land/std@0.144.0/node/internal_binding/cares_wrap.ts:412:23

TS2694 [ERROR]: Namespace 'Deno' has no exported member 'NAPTRRecord'.
      const records = (ret as Deno.NAPTRRecord[]).map(
                                   ~~~~~~~~~~~
    at https://deno.land/std@0.144.0/node/internal_binding/cares_wrap.ts:413:36

TS2345 [ERROR]: Argument of type '"NS"' is not assignable to parameter of type 'RecordType'.
    this.#query(name, "NS").then(({ code, ret }) => {
                      ~~~~
    at https://deno.land/std@0.144.0/node/internal_binding/cares_wrap.ts:431:23

TS2345 [ERROR]: Argument of type '"SOA"' is not assignable to parameter of type 'RecordType'.
    this.#query(name, "SOA").then(({ code, ret }) => {
                      ~~~~~
    at https://deno.land/std@0.144.0/node/internal_binding/cares_wrap.ts:451:23

TS2694 [ERROR]: Namespace 'Deno' has no exported member 'SOARecord'.
          ret[0] as Deno.SOARecord;
                         ~~~~~~~~~
    at https://deno.land/std@0.144.0/node/internal_binding/cares_wrap.ts:456:26

Found 18 errors.
```

However, when I deploy my **canary.11** version to netlify, the deploy logs show `[11ty] Copied 87 files / Wrote 131 files in 18.04 seconds (137.7ms each, v2.0.0-canary.12)`. Since it's **deploying with canary.12** then I thought it could be a local issue. So I deployed with GitHub Codespaces and canary.12 works as expected.

Regarding the TS2345 [ERROR], the only hint I have from Googling is [A beginners guide to HTTP server in Deno | Deno World](https://medium.com/deno-the-complete-reference/a-beginners-guide-to-http-server-in-deno-7b42bfcbbd0). 
```
The following is an example of an HTTP server that listens on port 5000 with an empty callback handler. This raise a compile-time error as the handler must return a response.

//Code
serve(() => {}, { port: 5000 });
//Test
$ deno run app.ts
error: TS2345 [ERROR]: Argument of type '() => void' is not assignable to parameter of type 'Handler'.
  Type 'void' is not assignable to type 'Response | Promise<Response>'.
Let’s update the code to return an empty response object. This would send a 200 OK without data.
```
**To Reproduce**
I don't know ¯\_(ツ)_/¯

**Expected behavior**
To execute without those errors.

**Workaround**
I moved my local dev environment to GitHub Codespace running on a 4-core with 8GB of memory and all is well again! Maybe it's the universe's way of telling me it's time to move on :-)

**Bottom Line** 
It's not you, it's me.

**Environment:**

- OS and Version: Mac 10.13.6
- Eleventy Version: 2.0.0-canary.12

**Additional context**
I wanted to post the issue in case someone else is experiencing a similar problem and are able to provide greater insight.

**Comment by Snapstromegon** [comment]
Hey, what is your deno version? Is it possible that it might be out of date?

**Comment by zachleat** [comment]
I’ll have a look at this but the first thing I’d recommend is trying to downgrade your `eleventyEdgeVersion`, see the advanced options here: https://www.11ty.dev/docs/plugins/edge/#2.-add-to-your-configuration-file

**Comment by zachleat** [comment]
`eleventyEdgeVersion: "1.0.1"` was the previous value. This will revert the behavior for https://github.com/11ty/eleventy/issues/2422

**Comment by stevenmilstein** [comment]
`eleventyEdgeVersion: "1.0.1"` Works

Thank you!

**Comment by stevenmilstein** [comment]
Is it possibly related to the "Deno for Visual Studio Code extension"? I let [Y/n] it install after `netliy cli` installed and have since removed it. Now I'm back canary.12 without downgrading `eleventyEdgeVersion: "1.0.1"`

**Comment by zachleat** [comment]
I don’t think so! I think your import maps were probably cached in the `.netlify` folder for `eleventy:edge`.

You can test this directly by changing your edge function import to bypass the import map like this:

```js
import { EleventyEdge } from "https://cdn.11ty.dev/edge@1.0.1/eleventy-edge.js"; // works
import { EleventyEdge } from "https://cdn.11ty.dev/edge@2.0.0/eleventy-edge.js"; // broken
import { EleventyEdge } from "https://cdn.11ty.dev/edge@2.0.1/eleventy-edge.js"; // works again
```

Regardless, this will be fixed by default in 2.0.0-canary.13, which will point to `edge@2.0.1`!

Thanks!

**Comment by stevenmilstein** [comment]
Yes, that works. 

Thank you
